sway/commands.c

```diff
@@ -1241,6 +1243,42 @@ static struct cmd_results *cmd_log_colors(int argc, char **argv) {
 	return cmd_results_new(CMD_SUCCESS, NULL, NULL);
 }
 
+static struct cmd_results *cmd_for_window(int argc, char **argv) {
+	struct cmd_results *error = NULL;
+	if ((error = checkarg(argc, "for_window", EXPECTED_AT_LEAST, 2))) {
+		return error;
+	}
+
+	// add command to a criteria/command pair that is run against views when they appear.
+	char *criteria = argv[0], *cmdlist = join_args(argv+1, argc-1);
+
+	struct criteria *crit = malloc(sizeof(struct criteria));
+	crit->crit_raw = strdup(criteria);
+	crit->cmdlist = cmdlist;
+	crit->tokens = create_list();
+	char *err_str = extract_crit_tokens(crit->tokens, crit->crit_raw);
+
+	if (err_str) {
+		error = cmd_results_new(CMD_INVALID, "for_window", err_str);
+		free(err_str);
+		free_criteria(crit);
+
+	} else if (crit->tokens->length == 0) {
+		error = cmd_results_new(CMD_INVALID, "for_window", "Criteria is empty");
+		free_criteria(crit);
+
+	} else if (list_seq_find(config->criterias, criteria_cmp, crit) != -1) {
+		sway_log(L_DEBUG, "for_window: Duplicate, skipping.");
+		free_criteria(crit);
+
+	} else {
+		sway_log(L_DEBUG, "for_window: '%s' -> '%s' added", crit->crit_raw, crit->cmdlist);
+		list_add(config->criterias, crit);
+	}
```

Comment:
style nitpicks - drop the extra newlines and make sure there are spaces around operators (i.e. `+`)